### PR TITLE
Image support in wysiwyg

### PR DESCRIPTION
**Pull request by jpuri**
Support for image upload in editor as detailed here: https://github.com/tinacms/tinacms/issues/166

### CODE REVIEW

File: packages/react-tinacms-editor/src/node-views/ImageView/index.ts [hunk 1]
```diff
@@ -35,7 +42,7 @@ export class ImageView implements NodeView {
     this.dom.classList.add('tinacms-image-wrapper')
     this.img = document.createElement('img')
     const { src, align, alt, title, width, height } = node.attrs
-    this.img.src = src
+    this.img.src = previewUrl(src)
```
Comment (jpuri) [inline packages/react-tinacms-editor/src/node-views/ImageView/index.ts]: NodeView will use `previewUrl` to get url for image. Default will be identity function.

File: packages/react-tinacms-editor/src/state/plugins/Menu/Image/ImageMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  if (!imageUpload) return null
+
+  const uploadImageFn = () => {
+    const imagePromises = imageUpload()
+    imagePromises.forEach(promise => {
+      promise.then(url => {
+        const { state, dispatch } = editorView.view
+        insertImage(state, dispatch, url)
+      })
+    })
```
Comment (jpuri) [inline packages/react-tinacms-editor/src/state/plugins/Menu/Image/ImageMenu.tsx]: Images will be inserted as promises resolve.

File: packages/react-tinacms-editor/stories/wysiwyg.stories.tsx [hunk 3]
```diff
@@ -67,6 +67,16 @@ const WithImage = () => {
           setValue(val)
         },
       }}
```
Comment (jpuri) [inline packages/react-tinacms-editor/stories/wysiwyg.stories.tsx]: Storybook example.

**Comment by jpuri** [comment]
@spbyrne : after these changes we also need a loading state in editor that can be shown while images are uploaded. Can you plz share ur suggestions.

**Comment by DirtyF** [comment]
Hey, works fine in Storybook. Should we update demos in a separate PR?

File: packages/react-tinacms-editor/src/state/plugins/Menu/index.tsx [hunk 4]
```diff
@@ -28,6 +28,7 @@ interface MenuProps {
   bottom?: boolean
   theme?: any
   sticky?: boolean
+  imageUpload?: () => [Promise<string>]
```
Comment (ncphillips) [inline packages/react-tinacms-editor/src/state/plugins/Menu/index.tsx]: I know this is still a draft but I just wanted to point this one out
```suggestion
  imageUpload?: (files: File[]) => [Promise<string>]
```

**Comment by jpuri** [comment]
@ncphillips : we can probably merge this one as its complete, I can work on drag-drop images in separate PR.

**Pull request merged**